modules/basics/src/main/java/com/opengamma/strata/basics/schedule/SchedulePeriod.java

```diff
@@ -227,19 +227,21 @@ public boolean contains(LocalDate date) {
    * The frequency and roll convention are used to build unadjusted schedule dates.
    * The stub convention is used to handle any remaining time when the new frequency
    * does not evenly divide into the period.
-   * 
+   *
    * @param frequency  the frequency of the sub-schedule
    * @param rollConvention  the roll convention to use for rolling
    * @param stubConvention  the stub convention to use for any excess
    * @param adjustment  the business day adjustment to apply to the sub-schedule
+   * @param combinePeriodIfNecessary whether to combine periods
    * @return the sub-schedule
    * @throws ScheduleException if the schedule cannot be created
    */
   public PeriodicSchedule subSchedule(
       Frequency frequency,
       RollConvention rollConvention,
       StubConvention stubConvention,
-      BusinessDayAdjustment adjustment) {
+      BusinessDayAdjustment adjustment,
+      boolean combinePeriodIfNecessary) {
```

Comment:
Strata needs backwards compatibility in places like this. You'll have to overload. I'm also not clear why the flag needs to be in here.